crates/libgraphql-core/src/test/golden_tests/test_runner.rs

```diff
@@ -0,0 +1,632 @@
+                    .collect();
+
+                let file_path = test_case.schema_paths[0].clone();
+                let snippet = create_missing_error_snippet(&file_path, &test_case.schema_expected_errors).ok();
+
+                GoldenTestResult {
+                    test_name,
+                    passed: false,
+                    error_message: Some(format!(
+                        "Expected: All error patterns must match\nGot: Not all expected errors matched\n\nUnmatched patterns:\n{}\n\nActual error:\n{error_str}",
+                        unmatched.iter().map(|p| format!("  ✗ {p}")).collect::<Vec<_>>().join("\n")
+                    )),
+                    file_path,
+                    file_snippet: snippet,
+                }
+            }
+        }
+    }
+}
+
+/// Extract a code snippet from a file with line numbers and error markers
+fn extract_snippet_with_error_marker(
+    file_path: &Path,
+    context_lines: usize,
+) -> Result<String, std::io::Error> {
+    let content = fs::read_to_string(file_path)?;
+    let lines: Vec<&str> = content.lines().collect();
+
+    let start_line = 0;
+    let end_line = context_lines.min(lines.len());
+
+    let mut snippet = String::new();
+    let line_num_width = (end_line + 1).to_string().len();
+
+    for (idx, line) in lines[start_line..end_line].iter().enumerate() {
+        let line_num = start_line + idx + 1;
+        snippet.push_str(&format!("   {line_num:>line_num_width$} │ {line}\n"));
+    }
+
+    Ok(snippet)
+}
+
+/// Create a snippet showing where an error was expected but didn't occur
+fn create_missing_error_snippet(
+    file_path: &Path,
+    _expected_errors: &[String],
+) -> Result<String, std::io::Error> {
+    let content = fs::read_to_string(file_path)?;
+    let lines: Vec<&str> = content.lines().collect();
+
+    let mut snippet = String::new();
+    snippet.push_str("   Expected errors based on comments:\n");
+
+    for (idx, line) in lines.iter().enumerate().take(10) {
+        let line_num = idx + 1;
+        if line.trim_start().starts_with("# EXPECTED_ERROR:") {
+            snippet.push_str(&format!("   {line_num:>3} → {line} ⚠️ (error not raised!)\n"));
+        } else if idx < 5 {
+            snippet.push_str(&format!("   {line_num:>3} │ {line}\n"));
+        }
+    }
+
+    Ok(snippet)
+}
+
+/// Run all operation validation golden tests
+pub fn run_operation_tests(fixtures_dir: &Path) -> GoldenTestResults {
+    let test_cases = GoldenTestCase::discover_all(fixtures_dir);
+
+    #[cfg(test)]
+    let test_results: Vec<GoldenTestResult> = test_cases
+        .par_iter()
+        .filter(|test_case| test_case.schema_should_be_valid)
+        .filter_map(|test_case| {
+            // Build the schema first
+            let schema = try_build_schema(&test_case.schema_paths)?;
+
+            // Test valid operations
+            let mut results = test_valid_operations(test_case, &schema);
```

Comment:
Since we'll use the same schema for each of these operations, can't we build the schema once and then share it among each of the operation tests?

Also: If the schema fails to build in any way during an operation test, we don't need to run any of the operation tests... And we should give a single error indicating the schema failed to build